apps/console/src/features/applications/components/application-danger-zone.tsx

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { hasRequiredScopes } from "@wso2is/core/helpers";
+import { AlertLevels, IdentifiableComponentInterface, SBACInterface } from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import { ConfirmationModal, DangerZone, DangerZoneGroup } from "@wso2is/react-components";
+import React, { FunctionComponent, ReactElement, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { AppConstants, AppState, FeatureConfigInterface, UIConfigInterface, history } from "../../core";
+import { deleteApplication } from "../api";
+   
+/**
+ * Prop types of the `LoginPlaygroundWizard` component.
+ */ 
+interface ApplicationDangerZonePropsInterface extends 
+    SBACInterface<FeatureConfigInterface>, IdentifiableComponentInterface {
+     /**
+      * Currently editing application id.
+      */
+     appId?: string;
+     /**
+      * Name of the application.
+      */
+     name: string;
+     /**
+      * Customized content.
+      */
+     content?: string;
+    }
+    
+/**
+ * Application Danger Zone component
+ *
+ * @return {JSX.Element}
+ */
+export const ApplicationDangerZoneComponent: FunctionComponent<ApplicationDangerZonePropsInterface> = (
+    props: ApplicationDangerZonePropsInterface
+): ReactElement => {
+    const {
+        featureConfig,
+        appId,
+        name,
+        content,
+        [ "data-componentid" ]: testId
+    } = props;
+     
+    const allowedScopes: string = useSelector((state: AppState) => state?.auth?.allowedScopes);
+    const UIConfig: UIConfigInterface = useSelector((state: AppState) => state?.config?.ui);
+ 
+    const [ showDeleteConfirmationModal, setShowDeleteConfirmationModal ] = useState<boolean>(false);
+    const [ isDeletionInProgress, setIsDeletionInProgress ] = useState<boolean>(false);
+ 
+ 
+    const { t } = useTranslation();
```

Comment:
Move hooks to the top just after props.

```tsx
    const {
        featureConfig,
        appId,
        name,
        content,
        [ "data-componentid" ]: testId
    } = props;
    
    const dispatch = useDispatch();
    
    const { t } = useTranslation();

```

Proper hierachy.

```
props
custom hooks
redux selctors
usestate
useEffect
functions
return
```

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/3042/commits/39901f546e17bf0fae6e1dfba50f7c76d0369360